src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -337,69 +338,78 @@ internal enum DS
         ////////////////////////////////////////////////////////////////////////////
 
         // End        NumEnd      NumAmPm     NumSpace    NumDaySep   NumTimesep  MonthEnd    MonthSpace  MonthDSep   NumDateSuff NumTimeSuff     DayOfWeek     YearSpace   YearDateSep YearEnd     TimeZone   Era         UTCTimeMark
-        private static readonly DS[][] s_dateParsingStates = [
+        private static ReadOnlySpan<DS> DateParsingStates => [
```

Comment:
Does making it a local in GetParsingState() improve anything?
(nit: `[` on its own line)

Reply:
Ah, ok. Previously I remember making such one-use props local vars (e.g. https://github.com/dotnet/runtime/commit/976409d1cf1997b30ba8844e18210e63ce3878f6) as overall it slightly produces cleaner code. [static](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGAHKAJYA3bBhhNySBgCUY2ACYB5AHYAbAJ4BlXtmUAeYOrEA+BgDMIEBgF5TAbQAMCB2iflXCUgF0A3Fx70DIZiDACyABQCyhgMAgCUHP48AQDs5pZ2Ar5JDAC+NLlAA==) vs. [local](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHgBPDDAYBZABQBLAHYYGEgJQcuPFQCUY2ACYB5KQBt+AZQAO2KQB4BQgHwMAZhAgMAvAwDaABgQe0X8r4RSAF0AbmUVbmIAdntHNwlQ8IYAXxpkoA)

Reply:
I don't think it's a big deal tbh, I just don't want to move a huge table inside a method - it effectively increases method body a lot => harder to read where it starts/ends? I'd also have to move `GetParsingState` method to become a local function

Reply:
> I don't think it's a big deal tbh

Ok.

> I'd also have to move GetParsingState method to become a local function

I think you meant instance method? I'm not sure why this would've been necessary; static method with local var collection expression would emit RVA-without-reflection-metadata same as it would for instance method. 👀